regulations/views/partial_search.py

```diff
@@ -83,16 +85,25 @@ def get_context_data(self, **kwargs):
             result['header'] = node_types.label_to_text(result['label'])
             if 'title' in result:
                 result['header'] += ' ' + title(result['title'])
-            result['section_id'] = section_url.view_label_id(
-                result['label'], context['version'])
-            result['url'] = section_url.fetch(
-                result['label'], context['version'], sectional=True)
+            if doc_type == 'cfr':
+                result['section_id'] = section_url.view_label_id(
+                    result['label'], context['version'])
+                result['url'] = section_url.fetch(
+                    result['label'], context['version'], sectional=True)
+            else:
+                result['section_id'] = '-'.join(
+                    [result['label'][0], 'preamble'] + result['label'])
+                result['url'] = reverse(
+                    'chrome_preamble',
+                    kwargs={'paragraphs': '/'.join(result['label'][:2])},
+                )
         context['results'] = results
 
-        for version in fetch_grouped_history(context['regulation']):
-            for notice in version['notices']:
-                if notice['document_number'] == context['version']:
-                    context['version_by_date'] = notice['effective_on']
+        if doc_type == 'cfr':
+            for version in fetch_grouped_history(context['regulation']):
+                for notice in version['notices']:
+                    if notice['document_number'] == context['version']:
+                        context['version_by_date'] = notice['effective_on']
```

Comment:
It seems like there are two largely different code paths here (one for 'cfr', one not). Would this be cleaner if they were pulled out into separate functions?